test/metabase/lib/field_test.cljc

```diff
@@ -182,13 +182,17 @@
                 (is (=? {:stages
                          [{:lib/type     :mbql.stage/mbql
                            :source-table "card__1"
-                           :breakout     [[:field {:join-alias "Products"} "CATEGORY"]]}]}
+                           :breakout     [[:field
+                                           {:join-alias (symbol "nil #_\"key is not present.\"")}
+                                           "CATEGORY"]]}]}
                         query'))
                 (is (=? [{:name              "CATEGORY"
-                          :display-name      "Category"
+                          :display-name      (if (:result-metadata card-def)
+                                               "Products → Category"
+                                               "Category")
```

Comment:
We keep using the display-name if it's available in the result metadata. If it's not, then we get just the name, since there is no join-alias.